src/plots/cartesian/axes.js

```diff
@@ -3780,12 +3785,33 @@ axes.drawLabels = function(gd, ax, opts) {
                 var pad = !isAligned ? 0 :
                     (ax.tickwidth || 0) + 2 * TEXTPAD;
 
-                var rotate90 = (tickSpacing < maxFontSize * 2.5) || ax.type === 'multicategory' || ax._name === 'realaxis';
-
-                // any overlap at all - set 30 degrees or 90 degrees
+                // old behavior for backwards-compatibility
+                var newAngle = ((tickSpacing < maxFontSize * 2.5) || ax.type === 'multicategory' || ax._name === 'realaxis') ? 90 : 30;
+                // autotickangles
+                if(autoTickAnglesRadians !== undefined) {
+                    var adjacent = tickSpacing;
+                    var opposite = maxFontSize * 1.25 * maxLines;
+                    var hypotenuse = Math.sqrt(Math.pow(adjacent, 2) + Math.pow(opposite, 2));
+                    var maxCos = adjacent / hypotenuse;
+                    var angleRadians = autoTickAnglesRadians.find(function(angle) { return Math.abs(Math.cos(angle)) <= maxCos; });
+                    if(angleRadians === undefined) {
+                        // no angle with smaller cosine than maxCos, just pick the angle with smallest cosine
```

Comment:
We talked about this with @alexcjohnson yesterday. 
It would be better to we simply pick the first one in the array that satisfy the condition.

Reply:
Do you mean, after it fails to find an angle whose cosine is ≤ `maxCos`, it should just pick the first angle in the list?